artix7/make/project_xray.cmake

```diff
@@ -1,86 +1,248 @@
   list(APPEND ${var} ${other_sources})
   set(${var} ${${var}} PARENT_SCOPE)
 endfunction()
 
-function(PROJECT_XRAY_INT)
+function(PROJECT_XRAY_DUMMY_SITE)
   set(options)
-  set(oneValueArgs PART INT)
-  set(multiValueArgs)
+  set(oneValueArgs PART SITE)
+
   cmake_parse_arguments(
-    PROJECT_XRAY_INT
+    PROJECT_XRAY_DUMMY_SITE
     "${options}"
     "${oneValueArgs}"
     "${multiValueArgs}"
     ${ARGN}
   )
 
-  string(TOLOWER ${PROJECT_XRAY_INT_INT} INT)
+  string(TOLOWER ${PROJECT_XRAY_DUMMY_SITE_SITE} SITE)
 
   get_target_property_required(PYTHON3 env PYTHON3)
   get_target_property(PYTHON3_TARGET env PYTHON3_TARGET)
 
-  set(INT_IMPORT ${symbiflow-arch-defs_SOURCE_DIR}/artix7/utils/prjxray-int-import.py)
-  get_project_xray_dependencies(DEPS ${PROJECT_XRAY_INT_PART} ${INT})
+  set(DUMMY_SITE_IMPORT ${symbiflow-arch-defs_SOURCE_DIR}/artix7/utils/prjxray_generate_dummy_site.py)
+  get_project_xray_dependencies(DEPS ${PROJECT_XRAY_DUMMY_SITE_PART} ${SITE})
+
   add_custom_command(
-    OUTPUT ${INT}.pb_type.xml
-    COMMAND ${PYTHON3} ${INT_IMPORT}
-      --part ${PROJECT_XRAY_INT_PART}
-      --tile ${PROJECT_XRAY_INT_INT}
-      --output-pb-type ${CMAKE_CURRENT_BINARY_DIR}/${INT}.pb_type.xml
+    OUTPUT ${SITE}.pb_type.xml ${SITE}.model.xml
+    COMMAND ${CMAKE_COMMAND} -E env PYTHONPATH=${PRJXRAY_DIR}:${symbiflow-arch-defs_SOURCE_DIR}/utils
+    ${PYTHON3} ${DUMMY_SITE_IMPORT}
+    --part ${PROJECT_XRAY_DUMMY_SITE_PART}
+    --site_type ${PROJECT_XRAY_DUMMY_SITE_SITE}
+    --output-pb-type ${CMAKE_CURRENT_BINARY_DIR}/${SITE}.pb_type.xml
+    --output-model ${CMAKE_CURRENT_BINARY_DIR}/${SITE}.model.xml
     DEPENDS
-      ${INT_IMPORT}
+    ${DUMMY_SITE_IMPORT}
       ${DEPS}
       ${PYTHON3} ${PYTHON3_TARGET}
     )
 
-  add_file_target(FILE ${INT}.pb_type.xml GENERATED)
+  add_file_target(FILE ${SITE}.pb_type.xml GENERATED)
+  add_file_target(FILE ${SITE}.model.xml GENERATED)
 endfunction()
 
-function(PROJECT_XRAY_CLB)
+function(PROJECT_XRAY_TILE)
   set(options)
-  set(oneValueArgs PART CLB)
-  set(multiValueArgs)
+  set(oneValueArgs PART TILE)
+  set(multiValueArgs SITE_TYPES)
   cmake_parse_arguments(
-    PROJECT_XRAY_CLB
+    PROJECT_XRAY_TILE
     "${options}"
     "${oneValueArgs}"
     "${multiValueArgs}"
     ${ARGN}
   )
 
-  string(TOLOWER ${PROJECT_XRAY_CLB_CLB} CLB)
+  string(TOLOWER ${PROJECT_XRAY_TILE_TILE} TILE)
 
   get_target_property_required(PYTHON3 env PYTHON3)
   get_target_property(PYTHON3_TARGET env PYTHON3_TARGET)
 
-  set(CLB_IMPORT ${symbiflow-arch-defs_SOURCE_DIR}/artix7/utils/prjxray-clb-import.py)
-  get_project_xray_dependencies(DEPS ${PROJECT_XRAY_CLB_PART} ${CLB})
+  set(TILE_IMPORT ${symbiflow-arch-defs_SOURCE_DIR}/artix7/utils/prjxray-tile-import.py)
+  get_project_xray_dependencies(DEPS ${PROJECT_XRAY_TILE_PART} ${TILE})
+
+  set(PART ${PROJECT_XRAY_TILE_PART})
+  set(PB_TYPE_INCLUDE_FILES "")
+  set(MODEL_INCLUDE_FILES "")
+  foreach(SITE_TYPE ${PROJECT_XRAY_TILE_SITE_TYPES})
```

Comment:
I think site types should be passed to the `tile_import` tool. That way it can check that the site type is valid for this tile type **and** skip site types which are not being provided.

Reply:
Agreed/